api/policies/checkDownloadToken.js

```diff
@@ -22,27 +22,23 @@
  * <http://www.gnu.org/licenses/>.
  */
 
-/* globals AccessToken */
-
-const sha1 = require("sha1");
+/* globals AccessToken,StorageService */
 
 module.exports = function(req, res, next) {
   let filename = req.query["filename"];
   let downloadToken = req.query["token"];
 
-  let timestamp = Date.now() / 1000;
-  let timeStone = timestamp + (3600 - timestamp % 3600);
-
-  AccessToken.findById(downloadToken.split(":")[0], (err, accessTokens) => {
+  AccessToken.findOne({
+    id: downloadToken.split(":")[0]
+  }, (err, accessToken) => {
     if (err !== null) {
       res.negotiate(err);
```

Comment:
`res.nogotiate` should be the last function. Please call with `return res.nogotiate`